src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -572,7 +619,7 @@ internal static void CreateSymbolicLink(string path, string pathToTarget, bool i
                 // If the handle fails because it is unreachable, is because the link was broken.
                 // We need to fallback to manually traverse the links and return the target of the last resolved link.
                 int error = Marshal.GetLastWin32Error();
-                if (IsPathUnreachableError(error))
+                if (IsPathUnreachableError(error) || error == Interop.Errors.ERROR_CANT_ACCESS_FILE /* Possibly broken AppExecLink */)
```

Comment:
The fact that the `IsPathUnreachableError` method is called in other places (specifically, the piece of code you shared), is the reason why I am placing the `ERROR_CANT_ACCESS_FILE` here, and not inside that method.

But the name of the method is to determine if a path is unreachable, so you're correct, I think it makes sense to catch `ERROR_CANT_ACCESS_FILE` there as well. I will try to add a unit test for `FileSystem.Attributes.Windows` that generates this error and catches it (hopefully, this can happen with an AppExecLink now).

I'll add the comment too.